clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -209,6 +222,32 @@ public EhcacheEntityResponse invokeAndWaitForRetired(EhcacheOperationMessage mes
     return invokeInternalAndWait(endpoint.beginInvoke().blockGetOnRetire(true), message, track);
   }
 
+  private void waitForInitCompleted(long nanos) {
+    boolean interrupted = false;
+    long deadlineTimeout = System.nanoTime() + nanos;
```

Comment:
Why playing with nanos and nanoTime if you are then converting to millis. Just use millis directly, The timeout doesn't need to be that precise.

Reply:
For relative time calculations System.nanotime is a bit more precise than System.currentTimeMillis (and in this particular case we have no need to know the exact time).